servant-server/src/Servant/Server/Internal.hs

```diff
@@ -173,7 +173,7 @@ instance (HasServer a context, HasServer b context) => HasServer (a :<|> b) cont
 -- > server = getBook
 -- >   where getBook :: Text -> Handler Book
 -- >         getBook isbn = ...
-instance (KnownSymbol capture, FromHttpApiData a
+instance (KnownSymbol capture, FromHttpApiData a, Typeable a
```

Comment:
This is a significant breaking change ; I am not sure we can require all captures to have a `Typeable` instance for debugging purposes.

Reply:
It's a pity that we didn't have it before but I don't know if we can afford having it not, indeed.

Reply:
Fair :) I feel old.

Reply:
I am guessing it still precludes `a` from being e.g. a function type, but maybe that's OK ?

Reply:
It would still break if someone wrote something like
```
genericAPI :: forall a. (FromHttpApiData a) => Proxy API
genericAPI = (Proxy :: Capture "x" a :> Get '[JSON] (Foo a)) 
```
They would need to add `Typeable a` to the constraints.